src/index.ts

```diff
@@ -0,0 +1,55 @@
+import TestCaseRunner from '@cucumber/cucumber/lib/runtime/test_case_runner';
+import {
+    PickleTag,
+    TestStep,
+    TestStepResult,
+    TestStepResultStatus,
+    StepKeywordType,
+    PickleStep
+} from '@cucumber/messages';
+
+type TestStepResultType = {
+    status: string,
+    type: string
+};
+
+declare global {
+    // eslint-disable-next-line no-var
+    var config: any;
+}
+
+const originIsSkippingSteps = TestCaseRunner.prototype.isSkippingSteps;
+// patch isSkippingStep method
+TestCaseRunner.prototype.isSkippingSteps = function () {
+    // @ts-ignore
+    const pickle = this.pickle;
+    // @ts-ignore
+    const testStepResults = this.testStepResults;
+    // @ts-ignore
+    const testCase = this.testCase;
+    // @ts-ignore
+    const currentTestStepId = this.currentTestStepId;
+
+    const activationTag = global?.config?.softAssertTag ?? '@softAssert';
```

Comment:
as I understand we will mark every step as `softAssert` in the scenario, am I right? 
even actions and memory steps will be marked as `softAsserp`?

Reply:
No, we need to mark only scenario with activation tag.
Login, in general, is following:
1. Check if scenario has activation tag
2. Check if step is Then (Outcome)
3. Check if there are no non-Then steps failed (IMO there are no sense to continue test if precondition or action failed)

if any of these checks failed - original implementation applied (step is skipped)